ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/multi-clusters-one-phz/provision/cucushift-installer-rehearse-aws-cases-multi-clusters-one-phz-provision-commands.sh

```diff
@@ -0,0 +1,527 @@
+# save hosted zone information to ${SHARED_DIR} for deprovision step
+echo "${HOSTED_ZONE_ID}" > "${SHARED_DIR}/hosted_zone_id"
+CHANGE_ID="$(echo "${HOSTED_ZONE_CREATION}" | jq -r '.ChangeInfo.Id' | awk -F / '{printf $3}')"
+
+# add a sleep time to reduce Rate exceeded errors
+sleep 120
+
+aws --region "${REGION}" route53 wait resource-record-sets-changed --id "${CHANGE_ID}" &
+wait "$!"
+echo "Hosted zone ${HOSTED_ZONE_ID} successfully created."
+
+# -----------------------------------------
+# Create install-config
+# -----------------------------------------
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
+pull_secret=$(<"${CLUSTER_PROFILE_DIR}/pull-secret")
+
+function create_install_config()
+{
+  local cluster_name=$1
+  local install_dir=$2
+
+  cat > ${install_dir}/install-config.yaml << EOF
+apiVersion: v1
+baseDomain: ${ROUTE53_HOSTED_ZONE_NAME}
+compute:
+- architecture: ${OCP_ARCH}
+  hyperthreading: Enabled
+  name: worker
+  platform: {}
+  replicas: 3
+controlPlane:
+  architecture: ${OCP_ARCH}
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+metadata:
+  creationTimestamp: null
+  name: ${cluster_name}
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  aws:
+    region: ${REGION}
+    hostedZone: ${HOSTED_ZONE_ID}
+    subnets: $(cat "${subnet_ids_file}")
+publish: Internal
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+
+  patch=$(mktemp)
+  if [[ ${CONTROL_PLANE_INSTANCE_TYPE} != "" ]]; then
+    cat > "${patch}" << EOF
+controlPlane:
+  platform:
+    aws:
+      type: ${CONTROL_PLANE_INSTANCE_TYPE}
+EOF
+    yq-go m -x -i ${install_dir}/install-config.yaml "${patch}"
+  fi
+
+  if [[ ${COMPUTE_NODE_TYPE} != "" ]]; then
+    cat > "${patch}" << EOF
+compute:
+- platform:
+    aws:
+      type: ${COMPUTE_NODE_TYPE}
+EOF
+    yq-go m -x -i ${install_dir}/install-config.yaml "${patch}"
+  fi
+}
+
+# -----------------------------------------
+# Generate proxy
+# -----------------------------------------
+
+proxy_public_url_file="${SHARED_DIR}/proxy_public_url"
+
+if [ ! -f "${proxy_public_url_file}" ]; then
+    echo "Did not found proxy setting from ${proxy_public_url_file}"
+    exit 1
+else
+    PUBLIC_PROXY_URL=$(< "${proxy_public_url_file}")
+fi
+
+if [ -z "${PUBLIC_PROXY_URL}" ]; then
+    echo "Empty proxy setting!"
+    exit 1
+else
+    cat > "${SHARED_DIR}/proxy-conf.sh" << EOF
+export http_proxy=${PUBLIC_PROXY_URL}
+export https_proxy=${PUBLIC_PROXY_URL}
+export no_proxy="localhost,127.0.0.1"
+export HTTP_PROXY=${PUBLIC_PROXY_URL}
+export HTTPS_PROXY=${PUBLIC_PROXY_URL}
+export NO_PROXY="localhost,127.0.0.1"
+EOF
+    cat > "${SHARED_DIR}/unset-proxy.sh" << EOF
+unset http_proxy
+unset https_proxy
+unset no_proxy
+unset HTTP_PROXY
+unset HTTPS_PROXY
+unset NO_PROXY
+EOF
+fi
+
```

Comment:
I guess we also can re-use `proxy-config-generate` instead of the lines of `Generate proxy` here, right?

Reply:
Update.